frontend/src/Components/HomePage/AddTrip/FlightForm.jsx

```diff
@@ -0,0 +1,242 @@
+import { useState, useEffect, useContext } from 'react';
+import { useNavigate } from 'react-router-dom';
+import axios from 'axios';
+import './FlightForm.css'
+import FlightResults from './FlightResults';
+import { API_BASE_URL } from '../../../config.js';
+import { UserContext } from '../../../UserContext';
+
+
+
+const FlightForm= () => {
+
+    const navigate = useNavigate();
+    const { user } = useContext(UserContext);
+    const [flightNumber, setFlightNumber] = useState(() => sessionStorage.getItem('flightNumber') || '');
+    const [departureDate, setDepartureDate] = useState(() => sessionStorage.getItem('departureDate') || '');
+    const [carrierCode, setCarrierCode] = useState(() => sessionStorage.getItem('carrierCode') || '');
+    const [flightStatus, setFlightStatus] = useState(() => {
+        const saved = sessionStorage.getItem('flightStatus');
+        return saved ? JSON.parse(saved) : null;
+    });
+    const [error, setError] = useState(null);
+    const [isLoading, setIsLoading] = useState(false);
+    const [flightAdded, setFlightAdded] = useState(() => sessionStorage.getItem('flightAdded') === 'true');
+    const [isAddingFlight, setIsAddingFlight] = useState(false);
+
+
+
+    useEffect(() => {
+
+        sessionStorage.setItem('flightNumber', flightNumber);
```

Comment:
Let's try to use context as suggested by Abe.

Reply:
Will do!